app/src/main/java/com/hedvig/app/feature/onbarding/ChoosePlanViewModel.kt

```diff
@@ -0,0 +1,58 @@
+package com.hedvig.app.feature.onbarding
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.hedvig.android.owldroid.graphql.ChoosePlanQuery
+import com.hedvig.android.owldroid.type.EmbarkStoryType
+import com.hedvig.app.feature.onbarding.ui.ChoosePlanActivity.Companion.COMBO
+import com.hedvig.app.feature.onbarding.ui.ChoosePlanActivity.Companion.CONTENTS
+import com.hedvig.app.feature.onbarding.ui.ChoosePlanActivity.Companion.TRAVEL
+import kotlinx.coroutines.launch
+
+abstract class ChoosePlanViewModel : ViewModel() {
+    protected val _data = MutableLiveData<Result<List<ChoosePlanQuery.EmbarkStory>>>()
+    val data: LiveData<Result<List<ChoosePlanQuery.EmbarkStory>>> = _data
+
+    abstract fun load()
+
+    private val _selectedQuoteType = MutableLiveData<OnboardingModel.Bundle>()
+    val selectedQuoteType: LiveData<OnboardingModel.Bundle> = _selectedQuoteType
+
+    fun setSelectedQuoteType(type: OnboardingModel.Bundle) {
+        _selectedQuoteType.postValue(type)
+    }
+
+    fun getWebPath() =
+        _selectedQuoteType.value?.embarkStory?.metadata?.find { metadata->
+            metadata.asEmbarkStoryMetaDataEntryWebUrlPath != null
+        }?.asEmbarkStoryMetaDataEntryWebUrlPath?.path
```

Comment:
I'm not quite sure that I understand this function 🤔

Reply:
It returns the url path for the selected embark story